### DISCUSSION TITLE
data for IGV issue

### DISCUSSION THREAD

**Discussion by laurel-nave-powers** [sr320/course-fish546-2021#102] [Q&A] [answered]
Hello, 
I'm not sure how I am supposed to visualize my data in IGV if I only have fastq files not fasta files. What should I use? 
Thanks!

**Comment by laurel-nave-powers** [comment, marked as answer]
I just tried to use a galaxy converter but it timed out and didn't work.

**Reply by laurel-nave-powers** [reply]
Now I am trying seqtk. I downloaded it using the instructions here (https://github.com/lh3/seqtk/blob/master/README.md) and tried the code to convert into fasta, but it said that seqtk could not be found. Not sure what I did wrong. Below is a screenshot of my terminal showing this. 
<img width="565" alt="Screen Shot 2021-02-26 at 9 06 30 AM" src="https://user-images.githubusercontent.com/76967600/109332014-6a1d9480-7812-11eb-8eff-b0ac1041a700.png">

**Reply by yaaminiv** [reply]
Have you tried using this? https://github.com/RobertsLab/code/blob/master/fasta.md#convert-fastq-to-fasta

`zcat input_file.fastq.gz | awk 'NR%4==1{printf ">%s\n", substr($0,2)}NR%4==2{print}' > output_file.fa`

**Reply by laurel-nave-powers** [reply]
I think that worked! Thank you!! 
A new issue now... IGV has let me upload the .fa file but it doesn't visualize anything. It is just blank.

**Reply by laurel-nave-powers** [reply]
Ok so maybe it didn't work. The .fa file is zero bytes, which would explain why it's showing up blank.

**Reply by kubu4** [reply]
> Now I am trying seqtk. I downloaded it using the instructions here (https://github.com/lh3/seqtk/blob/master/README.md) and tried the code to convert into fasta, but it said that seqtk could not be found. Not sure what I did wrong. Below is a screenshot of my terminal showing this.
> <img alt="Screen Shot 2021-02-26 at 9 06 30 AM" width="565" src="https://user-images.githubusercontent.com/76967600/109332014-6a1d9480-7812-11eb-8eff-b0ac1041a700.png">

You need to run the command like this:

```bash
./seqtk
```

Notice the `./`. This is required if you're running the command from within the directory that the program exists.

**Reply by laurel-nave-powers** [reply]
Thank you! That gave me an actual fasta file that isn't empty. Still coming up blank in IGV however. Am I missing something obvious? I watched the youtube video and am following along but am not getting the same results.

**Reply by kubu4** [reply]
Please post screen cap of IGV.

**Reply by laurel-nave-powers** [reply]
<img width="1123" alt="Screen Shot 2021-02-26 at 10 15 49 AM" src="https://user-images.githubusercontent.com/76967600/109338798-9ab5fc00-781b-11eb-9b70-2be3b575714a.png">

**Reply by kubu4** [reply]
Try making a selection in the dropdown menu _next_ to HIPPLAT menu.

**Reply by laurel-nave-powers** [reply]
It doesn't give me any options there.

**Reply by kubu4** [reply]
You'll need to wait awhile in order for it to load all the reads. The reason for the long delay is due to the fact that you've maxed out the available memory allocated to IGV (look in lower right corner of IGV for memory used and memory allocated). When memory is maxed out, it takes a long time for everything to get loaded (if it can even do it without crashing).

So, give it some time (10 - 15 mins?) and check it.

Alternatively, you can edit an IGV configuration file to allocate more memory, but it's going to be highly dependent on what you have available on your computer.

For reference, I converted a FastQ to FastA using `seqkit` like you did and loaded it into IGV. IGV is using ~11GB of memory to load that FastA file. FastA file size is 1.6GB.

**Reply by laurel-nave-powers** [reply]
Ohhh, ok that makes sense. That also might explain why my computer kept freezing when I try and use IGV. I will try again and give it some time. Thanks!

**Reply by kubu4** [reply]
Alternatively, use `seqtk` command to extract a small subset of sequences (just do 25 seqs so you can be sure to be able to visualize in IGV).

**Reply by laurel-nave-powers** [reply]
Using something like this? https://github.com/lh3/seqtk 
`seqtk subseq in.fq name.lst > out.fq`
I'm not sure how to make the name.lst that specifies what sequences to subset.

**Reply by kubu4** [reply]
Like this:

`./seqtk sample read1.fq 25 > sub1.fq`

**Reply by laurel-nave-powers** [reply]
I'm assuming I need to change it to use the fasta file instead of .fq to be able to open it in IGV?

**Reply by kubu4** [reply]
Run this:

`./seqtk sample read1.fq 25 > sub1.fq`

Then, this:

`./seqtk seq -A sub1.fq > sub1.fasta`

**Reply by kubu4** [reply]
So, yes, you need to convert to FastA before loading in IGV. Load the resulting `sub1.fasta` into IGV.

**Reply by laurel-nave-powers** [reply]
I think it worked! I am able to visualize it in IGV. Thank you so much for all your help!
